src/org/jetbrains/plugins/scala/lang/refactoring/introduceVariable/ScopeSuggester.scala

```diff
@@ -64,31 +78,34 @@ object ScopeSuggester {
       //check can we use upper scope
       noContinue = currentElement.calcType match {
         case projectionType: ScProjectionType
+          //we can't use typeAlias outside scope where it was defined
           if parent.asInstanceOf[ScTemplateBody].isAncestorOf(projectionType.actualElement) =>
           true
```

Comment:
We can move noContinue assign to here: noContinue = true.
We don't need to assign false.